src/Type/Php/StrlenFunctionReturnTypeExtension.php

```diff
@@ -37,11 +38,19 @@ public function getTypeFromFunctionCall(
 
 		$argType = $scope->getType($args[0]->value);
 
-		$constantStrings = TypeUtils::getConstantStrings($argType);
+		$constantScalars = TypeUtils::getConstantScalars($argType);
 		$min = null;
 		$max = null;
-		foreach ($constantStrings as $constantString) {
-			$len = strlen($constantString->getValue());
+		foreach ($constantScalars as $constantScalar) {
+			if ((new IntegerType())->isSuperTypeOf($constantScalar)->yes()) {
+				$len = strlen((string) $constantScalar->getValue());
+			} elseif ((new StringType())->isSuperTypeOf($constantScalar)->yes()) {
+				$len = strlen((string) $constantScalar->getValue());
+			} elseif ((new BooleanType())->isSuperTypeOf($constantScalar)->yes()) {
+				$len = strlen((string) $constantScalar->getValue());
```

Comment:
would `toString` also work here?
(let the type system decide whether the scalar is castable to string)

Reply:
`toString` returns `PHPStan\Type\Type`.
I don't see any way to get value from it, maybe I don't understand how to do it

Reply:
I am not sure about it, but I guess this could work:
```
$toStringType = $constantScalar->toString();
if ($toStringType instanceof ConstantScalarType) {
  $len = strlen($toStringType->getValue());
}
```